jwt/__init__.py

```diff
@@ -359,6 +359,17 @@ def load(jwt):
 
 def verify_signature(payload, signing_input, header, signature, key='',
                      verify_expiration=True, leeway=0, **kwargs):
+
+    if isinstance(leeway, datetime.timedelta):
+        try:
+            leeway.total_seconds
```

Comment:
At least in CPython, exceptions are only "expensive" when they actually occur, which would only be on Python 2.6 in this case, so I agree with @jpadilla that my approach using an exception is better than inspecting up front.

(In general, Python prefers EAFP over LBYL.)